docs/docs/advanced-use/transient-traits.md

```diff
@@ -0,0 +1,587 @@
+---
+title: Transient Traits and Identities
+---
+
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem';
+
+By default, Flagsmith [stores all Traits](/basic-features/managing-identities#identity-and-trait-storage) associated
+with an Identity to evaluate flags. This default behavior works for most use cases. However, there are scenarios where
+you may want more control over which Traits or Identities are stored. Using the Flagsmith API and SDKs, you can mark
+Traits and Identities as transient — meaning they are used for flag evaluation but are not stored.
+
+## Overview and Use Cases
+
+Transient Identities and Traits are particularly useful when you need to run flag evaluations without storing specific
+data in Flagsmith. Below are common scenarios where transient Traits or Identities come in handy.
+
+### Anonymous Identities
+
+If you choose to provide an empty or blank (`""`) identifier is provided, Flagsmith will automatically treat the
+Identity as transient. In this case, an auto-generated identifier is returned, based on a hash of the provided Traits.
+This ensures consistent flag evaluations without persisting the Identity in the Flagsmith dashboard.
+
+```javascript
+flagsmith.init({
+ evaluationContext: {
+  identity: {
+   identifier: null,
+   traits: { paymentPreference: 'cash' },
+  },
+ },
+});
+let identifierToUseLater = flagsmith.getContext().identity.identifier;
+```
+
+This approach is useful for scenarios such as A/B testing in e-commerce, where you want to include users who haven't
+registered or logged in.
+
+:::info
+
+Server-generated transient identifiers are consistent across flag evaluations, meaning the same identifier will be
+returned for identical sets of Traits.
+
+To generate truly unique identifiers, include additional unique Trait values.
+
+:::
+
+### Ephemeral Contexts
+
+In some cases, you may need to temporarily override certain Traits for a session or device without overwriting the
+stored value. For example, imagine you’ve persisted the `screenOrientation` Trait as `landscape` for a user:
+
+```javascript
+flagsmith.init({
+ evaluationContext: {
+  identity: {
+   identifier: 'my-user',
+   traits: { screenOrientation: 'landscape' },
+  },
+ },
+});
+```
+
+If you want to switch the `screenOrientation` to `portrait` for a specific session without overwriting the existing
+value, you can mark the Trait as transient:
+
+```javascript
+flagsmith.setTrait('screenOrientation', { value: 'portrait', transient: true });
+```
+
+In this case, `portrait` will be used for one specific flag evaluation, but the stored `landscape` value will remain
+unchanged.
```

Comment:
> one specific flag evaluation

I guess this will vary for different SDKs but this is a little unclear to me. Can we reword it to e.g. 'the next flag evaluation in the SDK, subsequent evaluations after that will use the previously stored value?' - is that even true? Wouldn't it remain as the transient value for the remainder of the 'session' ?

Reply:
> the next flag evaluation in the SDK, subsequent evaluations after that will use the previously stored value

TBH this sounds more confusing to me than what we have now

> Wouldn't it remain as the transient value for the remainder of the 'session' ?

How so? What even constitutes a session when we're talking about all possible SDKs?

Reply:
> > the next flag evaluation in the SDK, subsequent evaluations after that will use the previously stored value
> 
> TBH this sounds more confusing to me than what we have now
>
> > Wouldn't it remain as the transient value for the remainder of the 'session' ?
> 
> How so? What even constitutes a session when we're talking about all possible SDKs?

Well, my biggest issue is that the example you've given is using the `setTrait` method in the JS SDK. This doesn't directly retrieve the flags so I have no idea how it works. In other SDKs, it's pretty clear because you provide the traits directly when retrieving the flags.

Reply:
Fair enough. I've tried to improve the wording a bit.